data-prepper-plugins/decompress-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/decompress/DecompressionType.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.decompress;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.opensearch.dataprepper.model.codec.DecompressionEngine;
+import org.opensearch.dataprepper.plugins.codec.GZipDecompressionEngine;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public enum DecompressionType implements IDecompressionType {
```

Comment:
Please add some unit tests for the string to enum parsing. See `ForwardingAuthenticationTest` which does something similar for another enum. Also see that some tests use `@EnumSource` to help with future additions.

https://github.com/opensearch-project/data-prepper/blob/7e2331f1af4e646e49047de3bb90d642fc69dc46/data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/ForwardingAuthenticationTest.java#L24